apis/python/tests/test_sparse_nd_array.py

```diff
@@ -921,3 +922,58 @@ def test_create_platform_config_overrides(
     with tiledb.open(uri) as D:
         for k, v in expected_schema_fields.items():
             assert getattr(D.schema, k) == v
+
+
+def test_timestamped_ops(tmp_path):
+    # 2x2 array
+    with soma.SparseNDArray.create(
+        tmp_path.as_posix(),
+        type=pa.uint16(),
+        shape=(2, 2),
+        context=SOMATileDBContext(write_timestamp=10),
+    ) as a:
+        # write 1 into top-left entry @ t=10
+        a.write(
+            pa.SparseCOOTensor.from_scipy(
+                sparse.coo_matrix(([1], ([0], [0])), shape=a.shape)
+            )
+        )
+
+    # write 1 into bottom-right entry @ t=20
+    with soma.SparseNDArray.open(
+        tmp_path.as_posix(), mode="w", context=SOMATileDBContext(write_timestamp=20)
+    ) as a:
+        a.write(
+            pa.SparseCOOTensor.from_scipy(
+                sparse.coo_matrix(([1], ([1], [1])), shape=a.shape)
+            )
+        )
+
+    # read with no timestamp args & see both 1s
+    with soma.SparseNDArray.open(tmp_path.as_posix()) as a:
+        assert a.read().coos().concat().to_scipy().todense().tolist() == [
+            [1, 0],
+            [0, 1],
+        ]
+        assert a.nnz == 2
+
+    # read @ t=15 & see only the first write
+    with soma.SparseNDArray.open(
+        tmp_path.as_posix(), context=SOMATileDBContext(read_timestamp=15)
+    ) as a:
+        assert a.read().coos().concat().to_scipy().todense().tolist() == [
+            [1, 0],
+            [0, 0],
+        ]
+        assert a.nnz == 1
+
+    # read with (timestamp_start, timestamp_end) = (15, 25) & see only the second write
+    with soma.SparseNDArray.open(
+        tmp_path.as_posix(),
+        context=SOMATileDBContext(read_timestamp_start=15, read_timestamp=25),
+    ) as a:
+        assert a.read().coos().concat().to_scipy().todense().tolist() == [
+            [0, 0],
+            [0, 1],
+        ]
+        assert a.nnz == 1
```

Comment:
Rather than having each of the PIT read tests on individual objects, it could be worthwhile to place all of them together in a feature-specific test module, `test_point_in_time_semantics.py` (e.g.).  This makes it clear that we have many related tests for the various TileDBObjects that should be considered together when making future changes to PIT logic.  This would also allow us to DRY up the test code for the dense & spare array cases, by using a single pytest test method using `@pytest.mark.parametrize` to parameterize the classes/objects for the two cases.